lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -375,6 +375,49 @@ class DecomposeAtenAddmmOp : public OpRewritePattern<AtenAddmmOp> {
 };
 } // namespace
 
+// Decompose torch.mean into: sum(x)/div(numTensorElements).
+namespace {
+class DecomposeAtenMeanOp : public OpRewritePattern<AtenMeanOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMeanOp op,
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value input = op.self();
+    Value output = op.result();
+    BaseTensorType inputTensorType = input.getType().cast<BaseTensorType>();
+    BaseTensorType outputTensorType = output.getType().cast<BaseTensorType>();
+    // AtenDivScalarOp does not support non-floating type input.
+    if (IntegerType type = inputTensorType.getDtype().dyn_cast<IntegerType>())
+      return op.emitError("unimplemented: non-floating point type");
+
+    Value dtype = rewriter.create<ConstantNoneOp>(loc);
+    Value sum = rewriter.create<AtenSumOp>(loc, outputTensorType, input, dtype);
+    if (!sum)
+      return failure();
+
+    SmallVector<int64_t> sizes;
+    int64_t tensorSize = 1;
+    if (inputTensorType.hasSizes()) {
+      ArrayRef<int64_t> inputShape = inputTensorType.getSizes();
+      for (int64_t i : inputShape) {
+        if (i < 1)
+          return rewriter.notifyMatchFailure(op, "Invalid tensor size.");
+        tensorSize *= i;
+      }
+    } else
+      return rewriter.notifyMatchFailure(
```

Comment:
I think there were discussion about supporting dynamic cases. We could use `AtenSizeInt` to get the dynamic size of each dim and multiply together. We will need a `torch.aten.mul.int`.

Reply:
Yes, there were discussions, but in the end, we concluded that it would be better to use `torch.numel` for that. Once the support for `torch.numel` is added, I will add the support for dynamic cases.

Reply:
Ha, I see. Can we add support for `torch.numel` in the same PR? We could include multiple ops in the same PR if they are not too complicated and related.